reward/staking_info_db.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package reward
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type stakingInfoDB interface {
+	ReadStakingInfo(blockNum uint64) ([]byte, error)
+	WriteStakingInfo(blockNum uint64, stakingInfo []byte) error
+}
+
+func (sm *StakingManager) getStakingInfoDB(blockNum uint64) (*StakingInfo, error) {
+	if sm.stakingInfoDB == nil {
+		return nil, errors.New("stakingInfoDB is not set.")
+	}
+
+	jsonByte, err := sm.stakingInfoDB.ReadStakingInfo(blockNum)
+	if err != nil {
+		return nil, err
+	}
+
+	stakingInfo := new(StakingInfo)
+	err = json.Unmarshal(jsonByte, stakingInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return stakingInfo, nil
+}
+
+func (sm *StakingManager) addStakingInfoDB(stakingInfo *StakingInfo) error {
+	if sm.stakingInfoDB == nil {
+		return errors.New("stakingInfoDB is not set.")
+	}
+
+	marshaledStakingInfo, err := json.Marshal(stakingInfo)
+	if err != nil {
+		return err
+	}
+
+	err = sm.stakingInfoDB.WriteStakingInfo(stakingInfo.BlockNum, marshaledStakingInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
These are methods of `StakingManager`. We might move them to `reward/staking_manager.go` file declaring `StakingManager`.

Reply:
1. As you mentioned, the functions are receivers of `StakingManager` and it is good practice to put them in a file. However, the functions are actually for `stakingInfoDB`. Also, it will not be in consistency if `stakingInfoCache` has its own file and `stakingInfoDB` is not. Finally, I am working on refactoring `StakingManager` so that it will not have receiver. 
2. I applied your suggestion.

Reply:
Thank you for your kind reply.